qlib/TableMapper.qm

```diff
@@ -841,18 +919,53 @@ on_error table_mapper.rollback();
             remove rowcode;
         }
 
-        #! flushes queued data to the database
-        private hash flushIntern() {
+        /**
+         * flushes queued data to the database
+         * @param force_flush to flush even if there is not enough data to fill a block,
+         *                    e.g. on explicit flush() call
+         * @return Qore::NOTHING when nothing was flushed, otherwise returns
+         *         a bulk (hash of lists) with the data that were flushed into the DB,
+         *         updated by the 'ret_args' from the DB.
+         */
+
+        # return nothing if nothing needs to be flushed
+        private *hash flushIntern(bool force_flush) {
+            if (!exists hbuf) {
+                return; # nothing to flush
+            }
+            *int rec_list_size = getRecListSize(hbuf);
```

Comment:
`hbuf` is an internal buffer; it seems like another variable should be used to keep the buffer status instead of having to make a call to `getRecListSize()` on `hbuf` which seems to be inefficient